yt/yt/client/table_client/helpers.cpp

```diff
@@ -1568,4 +1569,61 @@ REGISTER_INTERMEDIATE_PROTO_INTEROP_BYTES_FIELD_REPRESENTATION(
 
 ////////////////////////////////////////////////////////////////////////////////
 
+TUnversionedValueRangeTruncationResult TruncateUnversionedValues(
+    TUnversionedValueRange values,
+    const TRowBufferPtr& rowBuffer,
+    const TUnversionedValueRangeTruncationOptions& options)
+{
+    std::vector<TUnversionedValue> truncatedValues;
+    truncatedValues.reserve(values.size());
+
+    int truncatableValueCount = 0;
+    i64 remainingSize = options.MaxTotalSize;
+    for (const auto& value : values) {
+        if (IsStringLikeType(value.Type)) {
+            ++truncatableValueCount;
+        } else {
+            remainingSize -= EstimateRowValueSize(value);
+        }
+    }
+
+    auto maxSizePerValue = std::max<i64>(0, remainingSize) / std::max(truncatableValueCount, 1);
+
+    i64 resultSize = 0;
+    bool clipped = false;
+
+    for (const auto& value : values) {
+        truncatedValues.push_back(value);
+        auto& truncatedValue = truncatedValues.back();
+
+        if (clipped || value.Type == EValueType::Any) {
+            truncatedValue = MakeUnversionedNullValue(value.Id, value.Flags);
+        } else if (value.Type == EValueType::Composite) {
+            if (auto truncatedCompositeValue = TruncateCompositeValue(TYsonStringBuf{value.AsStringBuf()}, maxSizePerValue)) {
```

Comment:
`TYsonStringBuf(value.AsStringBuf())` (we don't use curly braces for value constructors).